aws/resource_aws_vpn_gateway.go

```diff
@@ -206,13 +215,19 @@ func resourceAwsVpnGatewayAttach(d *schema.ResourceData, meta interface{}) error
 			if isAWSErr(err, "InvalidVpnGatewayID.NotFound", "") {
 				return resource.RetryableError(err)
 			}
+			if isAWSErr(err, "InvalidParameterValue", "This call cannot be completed because there are pending VPNs or Virtual Interfaces") {
```

Comment:
I would suggest moving this logic into `aws/config.go` so we can have the AWS Go SDK automatically retry this API call anywhere its used. See also: https://github.com/terraform-providers/terraform-provider-aws/blob/7cb3734472b25b4515367d55326b88ae480c5937/aws/config.go#L556-L574